src/lib/storage/fitted_attribute_vector.hpp

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include <memory>
+#include <vector>
+
+#include "../types.hpp"
+#include "../utils/assert.hpp"
+#include "base_attribute_vector.hpp"
+
+namespace opossum {
+template <typename T>
+class FittedAttributeVector : public BaseAttributeVector {
+ public:
+  explicit FittedAttributeVector(const size_t size) { _content = std::make_shared<std::vector<T>>(size); }
```

Comment:
You might want to make sure if the size fits the T max-value.

Reply:
Actually the desired length of the attribute vector `size` is completely independent of the width of the data type available to the elements. I believe there is no reliable information on the maximum length of a vector. When exceeding the limit, the constructor of the vector will likely throw an exception.